### DISCUSSION TITLE
[erledigt] Freetzen einer 6690

### DISCUSSION THREAD

**Discussion by Prisrak1** [Freetz-NG/freetz-ng#530] [push firmware] [answered]
habe mir eine 6690_07.39.ger-rev96142_labor_freetz-ng-34137MOA.image gebaut und wollte diese mit push firmware flashen. Mehre Versuche gingen fehl. 

Anschluss per Lankabel. Wlan /Firewall und Antivirus deaktiviert. ./tools/push_firmware /home/freetz/6x-ng/images/6690_07.39.ger-rev96142_labor_freetz-ng-34137MOA.image. Leider geht nicht. Vermute gibts kein FTP Zugriff. Das Anhalten in FTP geht auch mit fremden recovery nicht.
Netzwerkeinstellungen usw. wie bei pushen der F!B en 6490, 6590, 6660, wo es super funktionierte. Liegt es an der Bootloderversion? Es ist eine Freie Box.

Gehts bei jemanden überhaupt?

**Comment by fda77** [comment]
Wie immer bei neuen Geräten: Finde zuerst heraus ob es an Freetz liegt oder am flashen, beides hat noch niemand getestet.
Da es von der Box bislang immerhin/nur eine inhaus und eine labor gibt würde ich zuerst eine von diesen UNVERÄNDERT flashen um zu sehen ob das flashen funktioniert. Dann poste dazu aber die Ausgabe von push firmware oder den ftp-Befehlen die du nutzt, um zu sehen was die Fehlermeldung usw ist! "geht nicht" ist relativ

**Comment by Prisrak1** [comment]
habe "untrustedd" wie in 5530 auf no, Aktuell mit und ohne watchdog getested. Alles gleich. Lankabel ist nicht defekt. Auf den folgenden Befehl reagiert die Box nicht und startet durch.  Letzte geänderte Rev: 34161

```
freetz@freetz:~/6x-ng$ ./tools/push_firmware /home/freetz/6x-ng/images/6690_07.39.ger-rev96142_labor_freetz-ng-34161MA-UNKNOWN_20220504-161722.image -cmd ftp

* Analyzing '/home/freetz/6x-ng/images/6690_07.39.ger-rev96142_labor_freetz-ng-34161MA-UNKNOWN_20220504-161722.image' ...
 * Product: Fritz_Box_HW267a (FRITZ!Box 6690 Cable)
 * Using command: ftp
 * Target host: 192.168.178.1
 * Outgoing IP: 192.168.178.17
 * Flash mode: uimg-boot
 * Designated linux_fs_start: 1

 !!! WARNING !!! WARNING !!! WARNING !!! WARNING !!! WARNING !!!
 !!!  THERE IS NO WARRANTY AT ALL !!! USE AT YOUR OWN RISK   !!!

 * Are you sure, that you want to flash this file to the device?
   /home/freetz/6x-ng/images/6690_07.39.ger-rev96142_labor_freetz-ng-34161MA-UNKNOWN_20220504-161722.image
   Proceed? (y/[n]) y

 * You should now reboot your box (192.168.178.1). Waiting for shut down.
   Switch off, if reboot is not detected because it happens too quickly.
   Some newer bootloader versions allow to flash on power-cycle only...
```

![Screenshot 2022-05-04 190049](https://user-images.githubusercontent.com/45603872/166740811-0f7b5915-e75a-4b22-bd85-82a26e194e06.png)

**Comment by fda77** [comment]
Ah, das mit diesen 4 MTDs der UIMG Geräte...
Zuerst watchdog/untrusted ausmachen ist am anfang okay, es gibt aktuell aber nur 1 die das braucht, und die hat FIT, nicht UIMG.
Bei älteren Docsis Geräten waren das noch andere MTD "Nummern", diese mit "=>;<" sind die neueren: https://github.com/Freetz-NG/freetz-ng/blob/master/tools/push_firmware#L120=
Es könnte sein dass bei diesem Gerät wieder andere Nummer genutzt werden - die muss man noch herausfinden! Normal macht man ein recovery und schneidet es mit tcpdump mit - gibt es für docsis meist nicht. Daher müsste man im entpacketen Dateisystem herumsuchen, in den Sourcen (email schreiben!!) oder vielleicht findet man in den supportdaten was. Poste daraus doch mal alles mit Partitionen, mtd usw - oder einfach alles falls keine privaten Daten drin sind. 
Für die 6591 wurden die MTDs hier gesucht: https://bitbucket.org/fesc2000/ffritz/src/6591/README-6591.md

Ohne vorhandenes Recovery ist es übrigens sehr gefährlich das zu testen! Hoffentlich hast du mindestens in jeder "partition" ein Fritzos

Diese komischen Revisionen >30.000 sagen übrigens nichts aus da du von github via svn ausgecheckt hast. Wenn du von boxmatrix mit svn oder github mit git auscheckst sind die im Bereich 19xxx: https://github.com/Freetz-NG/freetz-ng#mirrors

**Reply by fda77** [reply]
Ich hab die 6690 vorerst deaktiviert: https://github.com/Freetz-NG/freetz-ng/commit/52e010ec881832e05bf729b87171690bf460a81f
Falls du trotzdem experimentieren möchtest, die Meldung wieder löschen

**Comment by Prisrak1** [comment]
ja, ich teste gern weiter und werde mitteilen, wenn was sich ergibt. Habe auch kein Problem, wenn´s schiefgeht.
Anfangszustand Auslieferung v 7.29. z.Z FRITZ!OS:[07.39-96251 Inhaus. Mit dem Zusatz "-ifs 1" wird versucht bei mir die "=>;<" zu beschreiben.  Mit dem Zusatz "-ifs 0" gehts mit "6701". Ergebnis gleich. Muss man wohl suchen.
Auf meine Email an AVM gabs heute ein Link. Wohin damit? habs heruntergeladen.

![grafik](https://user-images.githubusercontent.com/45603872/166887425-046d18c9-d296-48cd-b44a-ebf8958417c7.png)
Wie komme ich an die original 7.29 ran? Ich glaube, ich könnte mir jetzt was bauen.. war mal was so.  

oder z.B. von der 7.39, dieselbe flashen und auf "Laborversion. Zurück zum offiziellen FRITZ!OS". Dann da die beiden Partitionen mit 7.39 überschrieben sind, wird doch die 7.29 heruntergeladen. Mit Wireshark logen (woher kommt die zum Herunterladen), oder bleibt die irgendwo erhalten?

**Comment by fda77** [comment]
> Habe auch kein Problem, wenn´s schiefgeht.

Das könnte bei diesem Gerät im schlimmsten Fall "Briefbeschwerer" bedeuten!!!

"=>;<"  und  "6701" sind die Partitionen auf die versucht wird zu schreiben, die sind in PF hinterlegt. Sie sind hier evtl anders oder es musst gar mit anderen Befehlen geflasht werden (glaube ich nicht).
Jedenfalls wenn man eine Partition überschreibt in der was "wichtiges" drin ist könnte es für immer weg sein...

> Wie komme ich an die original 7.29 ran? .. "Laborversion. Zurück zum offiziellen FRITZ!OS"

Da müsste man wissen wie die Box die url findet. Optimal: du konfigurierst die 6690 als client im lan, und auf dem router schneidesz du mit tcpdump ausgehenden Traffic auf port 80 mit (-w datei.cap). Dann kann man sich da anschauen und nachbauen
Für normale Firmware geht da ja, siehe Diskussion juis_check https://github.com/Freetz-NG/freetz-ng/discussions/529:
``` 
6660:
env - tools/juis_check         HW=252  Flag=cable_retail Annex=Kabel  -s
Found newer version: 252.07.29

6690:
env - tools/juis_check         HW=267 Flag=cable_retail Annex=Kabel  -s
No newer version found.
``` 
"-i" statt "-s" zeigt die URL

> Auf meine Email an AVM gabs heute ein Link. Wohin damit? habs heruntergeladen.

Normal macht avm das auf http://osp.avm.de , da ist aber nichts.
Sind ja nur GPL Sachen, lad es irgendwo hoch

**Reply by fda77** [reply]
Hab die releas-firmware gefunden; `env - tools/juis_check --down  HW=267  Flag=cable_retail Annex=Kabel `

**Reply by Prisrak1** [reply]
was mache ich falsch, wo befindet sich der Skript yf (auf linux oder Fritz.Box) , bzw. aus welchem Verzeichnis gibst du den Befehl ein?

Bei mir  Ubuntu -> freetz@freetz:~/YourFritz$ env - tools/juis_check --down HW=267 Flag=cable_retail Annex=Kabel
kommt: -> juis_check: Unknown option '--down' specified.

oder bei : env - tools/juis_check         HW=267 Flag=cable_retail Annex=Kabel  -s
kommt: -> juis_check: Missing file name after option '-s'.

bei: -> env - tools/juis_check         HW=267 Flag=cable_retail Annex=Kabel  -i
kommt: -> juis_check: Missing FRITZ!Box IP address or DNS name.

**Reply by fda77** [reply]
Du benutzt das alte script von peterpawn, keine ahnung wie du das in freetz reinbekommen hast! Ist dein Checkout über 1 Jahr alt?
Das gleichnamige und parameterkompatible in Freetz ist wesentlich schneller und hat einiges mehr an Optionen (zb das "downgrade" das du benutzen willst)

**Reply by Prisrak1** [reply]
gestern und heute erneuert:  git clone https://github.com/PeterPawn/YourFritz.git YourFritz
gibts was anderes?

**Reply by fda77** [reply]
Steht doch da:
> Das gleichnamige und parameterkompatible in Freetz

Natürlich im Checkout des repos in dem du fragst!?! Verzeichnis+Name schrieb ich oben.
Zu PPs zeug kann ich dir nichts sagen, ist zu sperrig.

Damit wird [docs/juis/README.md ](https://github.com/Freetz-NG/freetz-ng/tree/master/docs/juis) generiert

**Reply by Prisrak1** [reply]
habs heute schon davor gelesen, nur nicht umgesetzt ..

**Comment by Prisrak1** [comment]
hier: 
[1](https://www.file-upload.net/download-14920579/source-files-FRITZ.Box_6690_Cable-armp7-07.29.tar.gz.html
)

[2](https://www.file-upload.net/download-14920581/source-files-FRITZ.Box_6690_Cable-atomp7-07.29.tar.gz.html
)

**Reply by fda77** [reply]
"atom" ist immer interessanter. mit der stable firmware musst du noch warten bis https://github.com/Freetz-NG/freetz-ng/actions/runs/2277178861 fertig ist

**Comment by Prisrak1** [comment]
edit.. habs gesehen...

**Reply by fda77** [reply]
Download war kein Problem, ging mit voller "Geschwindigkeit" soweit man das bei einer gammeligen Telekom Leitung so nennen kann. Für einen Filehoster ohne Anmeldung nicht schlecht. Nur nicht so gut dass das nur per Browser geht und nicht mit zb wget.

**Comment by fda77** [comment]
Wie gesagt, es kann sein dass du das Gerät in einen Briefbeschwer verwandelst, also auf dein Gefahr!

Es könnte sein dass die X86 und ARM Partitionen vertauscht sind. Also nicht
```
[ $LFS -ne 1 ] && ARMsysMTD='6'  && ARMkrnMTD='7'  && X86sysMTD='0'  && X86krnMTD='1'  ...
```
sondern
```
[ $LFS -ne 1 ] && ARMsysMTD='0'  && ARMkrnMTD='1'  && X86sysMTD='6'  && X86krnMTD='7'  ...
```
für lfs=0
Der Bootloader sieht die Partitionen anders als das laufende System

Kannst du noch Supportdaten von avm anhängen, oder alles daraus was mit partitionen, mtd usw zu tun hat?

**Comment by Prisrak1** [comment]
uptime: 17:08:22 up  7:08,  load average: 1.12, 1.16, 1.17
4.9.279
DMC	RTL=y,SL1,nonFPGA
HWRevision	267
HWSubRevision	1
ProductID	Fritz_Box_HW267
SerialNumber	
annex	Kabel
autoload	yes
bootloaderVersion	1.11534
country	049
firstfreeaddress	0x7D3C4C00
firmware_info	267.07.39
firmware_version	avm
language	de
linux_fs_start	1
..
memsize	0x40000000
mtd0	0x0,0x4800000
mtd1	0x4800000,0x5100000
mtd2	0x5100000,0x5300000
mtd3	0x5300000,0x5B00000
mtd4	0x0,0x0
mtd5	0x5B00000,0xDB00000
mtd6	0xDB00000,0xEE00000
mtd7	0xEE00000,0xF300000
mtd8	0x0,0x0
mtd10	0x0,0x0
mtd11	0xF300000,0x13B00000
mtd12	0x13B00000,0x14400000
mtd13	0x14400000,0x15700000
mtd14	0x15700000,0x15C00000
..

habe gemerkt, in der Supportdatei sind sehr viele Privatdaten dabei. Wenn du noch die nicht geladen hast, teile genauer, welche Zeilen dich interessieren.

**Reply by fda77** [reply]
Wie bereits in https://github.com/Freetz-NG/freetz-ng/discussions/530#discussioncomment-2689145 und https://github.com/Freetz-NG/freetz-ng/discussions/530#discussioncomment-2697234 geschrieben: alles mit mtd und partitionen

**Comment by Prisrak1** [comment]
kann aus meinem Verständnis nichts Passendes finden .. noch ein Tip?
 zu Filehoster und wget, was wäre besser?

**Reply by fda77** [reply]
Ich hab auch nichts sinnvolle gefunden, nur die Idee aus https://github.com/Freetz-NG/freetz-ng/discussions/530#discussioncomment-2697234
Einen besseren Hoster kenne ich nicht, Dropbox, Mega etc sind auch nicht besser. AVM wird die doch irgend wann wohl auf ihren Server packen...

**Comment by Prisrak1** [comment]
fesc meint " .. mtd0 0x0,0x4800000 → 72 MB → atom rootfs → mmcblk0p3
mtd1 0x4800000,0x5100000 → 9MB → atom kernel → mmcblk0p2
mtd6 0xDB00000,0xEE00000 → 19MB → arm rootfs → mmcblk0p5
mtd7 0xEE00000,0xF300000 → 5MB → arm kernel -> mmcblk0p4

Bleibt also alles wie gehabt. So man denn ftp zum laufen bekommt sollte es funktionieren. .." 

kannst wieder freischalten mit push. ich kann erst erneut am Wochenende testen.

**Reply by Prisrak1** [reply]
hat sich erledigt. Der Grund: meine Helferin hat den Lankabel schlecht reingesteckt gehabt. Somit, es geht.
![7 29](https://user-images.githubusercontent.com/45603872/203236478-b1e41349-ce2c-4c31-a5e7-17e517bd49b9.png)

**Reply by roland3939** [reply]
Nach 5 Monaten den Fehler gefunden 😄😄😄🙈🙈🙈🧑‍🦯🧑‍🦯🦮🦮

**Reply by Prisrak1** [reply]
manchmal vertraut man einfach so ohne zu hinterfragen.

**Comment by fda77** [comment]
Interessant. Und Freetz läuft auch? Häng doch mal ne "AVM-Info" Screenshot an. Es steht noch UNTESTED dran

**Reply by fda77** [reply]
9b33963e7

**Reply by Prisrak1** [reply]
Grundsätzlich läufts mit 7.29 und 7.39. Es gingen einige Sachen nicht. (Telefonie, Alexa und einige https://Webseiten) Habs z.Z. wieder auf Original zurückgesetzt. Bevor ich es nutze, würde ich es gerne weiter testen, ob´s bei mir liegt. ich muß abwarten, bis keiner bei mir zuhause meckert und dann teste ich weiter. Musste zudem auch per FTP flashen. Hats länger gedauert.
Wäre schön, wenn die Updatefunktion in Freetz wieder funktionieren würde,

**Reply by fda77** [reply]
Ich kann mir unter "Telefonie, Alexa und einige https:// webseiten" nicht wirklich was vorstellen, meld dich einfach wenn es (nicht) geht

**Reply by Prisrak1** [reply]
Habs noch mal getestet. 
Gebaut wurden die v. 2.29 und 7.39 mit gleichen Einstellungen. Nur bei der Version 7.29 funktionieren keine der https Verbindungen und keine Dectelefonie.

**Comment by Prisrak1** [comment]
Das Pushen funktioniert, jedoch die Box startet danach nicht selbständig. Die Bootpartition wechselt auch nicht selbstständig. Ich musste diese über FTP wechseln. Im System fehlt der Punkt "Firmware update" Sonst funktioniert alles  auch in Beta einwandfrei! Beide Partitionen lassen sich beschreiben. z.Z, hab auf einer 7.29 Original und auf der anderen 7.39 Freetz.
![Screenshot_20221126_095341](https://user-images.githubusercontent.com/45603872/204081159-d4a52e1e-8cb1-41bc-a3ff-e56bde0fd3cf.png)
![Screenshot_20221126_100839](https://user-images.githubusercontent.com/45603872/204081469-50834572-3502-4ea6-bd2f-98defcdf6f62.png)

**Reply by fda77** [reply]
>  Im System fehlt der Punkt "Firmware update

Das ist so weil jemand (evtl du???) gesagt hat dass es "nich geht". Was genau müsste jemand mit so einer Hardware (hab keine) herausfinden.

Das andere musst du/jmd mit hardware auch herausfinden/ausprobieren, ohne avm-recovery .exe bleibt sonst nichts

**Reply by Prisrak1** [reply]
wo und welche Zeilen muss ich aktivieren, um es bei mir zu testen?

**Reply by fda77** [reply]
Die Zeile löschen und es sollte wieder im Webif sein: https://github.com/Freetz-NG/freetz-ng/blob/bf1a5149ff865ff0474632fe456ae220a7989066/make/pkgs/mod/mod.mk#L29

Siehe auch https://github.com/Freetz-NG/freetz-ng/commit/d1c8224b5b0f70810ada57c848693ee0d9c7401c und https://www.ip-phone-forum.de/threads/313424

**Reply by Prisrak1** [reply]
(ffritz) .. und [...](https://www.ip-phone-forum.de/threads/313424) kenne ich schon und "igostischerweise" hoffte ich sehr, du machst es locker vom Hocker. :)
nun da es bei mir auch fläshbar ist, teste ich gerne weiter. 
Ps. Die Puma6/7-Modelle sind aus meiner Erfahrung unzerstörbar. Somit müsste es bei allen funktionieren. Danke für deine Hinweise! da braucht man kein Recovery.

**Reply by fda77** [reply]
Solang man den Bootloader nicht kaputt macht, kann man über ftp eigentlich immer wieder was flashen, bei allen. Environment (MAC) und Doscis Zertifikate am besten auch lassen wie sie sind.
Die Problembehebung kann recht einfach sein, es müsste aber mal jemand mit passender Hardware das Problem zeigen. Ich versteh eh nicht warum das ständig bei vielen so ein riesen Problem ist

**Comment by Prisrak1** [comment, marked as answer]
habs mit Installieren des Images (mit ssh/telnet/Konsolenzugriff)​ und über das "update in Freetz" bei den F!B 6660 / 6690 getestet. Funktioniert einwandfrei! Der Wechsel der Startpartition geht auch automatisch beim Update. jedoch in Freetz nur Wechsel der Startpartition ohne Update geht nicht. In Putty gehts mit "/bin/aicmd pumaglued uimg switchandreboot
"
![Screenshot_20221202_120325](https://user-images.githubusercontent.com/45603872/205286109-ed92a154-7c95-4bab-8695-b7be16b8d14a.png)

**Reply by fda77** [reply]
Cool, danke: 872ad353a Da hatte wohl mal wieder jemand ein faules Image gefalsht und dann ein "geht nicht" rausgehauen :-\
Die 3 Puma7 bzw UIMG 6591/6660/6690 sollten eigentlich recht gleich sein, hätte mich gewundert wenn die 6690 was ganz anderes wäre

**Comment by fda77** [comment]
Das mit dem Partitionen-Wechseln müsste man sich noch überlegen...
```
/bin/aicmd pumaglued uimg switchandreboot
```
Hört sich nach einem sofortigen Reboot... gibts da nicht was anderes?
Vielleicht den Button bei den UIMG einfach ausblenden?

Jedenfalls ist in make/pkgs/mod/files/root/usr/mww/cgi-bin/exec.d/linux_fs_start.sh das "echo" der Befehl für andere Geräte

**Comment by Prisrak1** [comment]
sofortiger Reboot.. "Partition wechseln" wird ja auch ohne den Wechsel der Startpartition in Freetz durchgeführt. Warum wechselt er nicht... kann nur am Befehl liegen. Die Infos zu "FritzOS UNKNOWN" werden auch nicht angezeigt.

**Comment by fda77** [comment]
"FritzOS UNKNOWN" ist was anderes, da müsste die inaktive Patition gemountet werden und untersucht. Irgendwas geht wohl nicht. Ohne UIMG Gerät hab ich aber keine Idee wo es hängt, das müsste jemand anderes herausfinden. Eigentlich dachte ich das ging

> "Partition wechseln" wird ja auch ohne den Wechsel der Startpartition in Freetz durchgeführt

Hä? Kapiere nicht was du meinst

**Reply by fda77** [reply]
Aber sag mal, ist im inaktiven Partitionsset überhaupt etwas sinnvolles drin?!?

**Comment by fda77** [comment]
@Prisrak1: Teste mal ob mit 8cbe9a094 die partition-wechseln geht

**Comment by Prisrak1** [comment]
![image](https://user-images.githubusercontent.com/45603872/205666823-df9dcbbb-9570-4cdf-ac35-d1f433660af6.png)
![image](https://user-images.githubusercontent.com/45603872/205840324-5cd60367-04bb-42ff-8f8d-c9b7618c9cb3.png)
getestet und wie man im unteren Bild sehen kann, bleibt es bei der Meldung und passiert nichts weiter. Es sind schon zwei verschiedene Versionen auf der F!B drauf.

**Reply by freetz-ng-mod** [reply]
Kann es sein das es an den Sachen liegen könnte
https://github.com/Freetz-NG/freetz-ng/blob/f925d87a10bbd008b74e6cb3de0aa40a6bdf33f9/config/ui/freetz.in#L233
oder
https://github.com/Freetz-NG/freetz-ng/blob/f925d87a10bbd008b74e6cb3de0aa40a6bdf33f9/config/ui/patches.in#L1300

Da mir auf der 7530AX auch nichts angezeigt wurde, habe ich da selber was geändert. Seit dem steht wieder was bei mir

**Reply by fda77** [reply]
> Seit dem steht wieder was bei mir

Glückwunsch, welch eine Bereicherung ^^

FIT Modelle haben hiermit nichts zu tun.
Ansonsten versteh ich unter Programmieren mehr etwas am Code zu ändern statt komische Konstellationen in der .config zu fabrizier. Finde doch heraus was das Problem ist

**Reply by freetz-ng-mod** [reply]
Der Fehler ist ja bekannt, schon länger. Aber es wird ja nicht geändert. Für die 7530AX gibt es ja kein loop in der make menuconfig.

**Reply by fda77** [reply]
Nö, ging als ich letztes mal ausprobierte. Bleibt denn ot

**Comment by fda77** [comment]
@Prisrak1
`Toggling` ging nicht wegen diesem Fehler: a9d019d2e, nochmal testen..
An `UNKNOWN`  wurde nichts verändert

**Reply by Prisrak1** [reply]
![image](https://user-images.githubusercontent.com/45603872/205972904-fbd8de11-7713-4d56-8282-ac486b88523e.png)
der Wechsel findet nicht statt.

**Reply by fda77** [reply]
Hm, ich blicke nicht so ganz durch was du da gemacht hast. Wieso manueller Neustart?
Wenn du die Partition wechselst sollte (neuerdings) diese Meldung kommen:
https://github.com/Freetz-NG/freetz-ng/blob/master/make/pkgs/mod/files/root/usr/mww/cgi-bin/exec.d/linux_fs_start.sh#L14
```echo "changing $LFS_LIVE -> $LFS_DEAD ... and rebooting"```

**Comment by Prisrak1** [comment]
ich drücke den Button "Firmwarepartition wechseln" und dann "Box neustarten" Reboot

**Comment by fda77** [comment]
Bei den UIMG wird "vermutlich" nach "Firmwarepartition wechseln" automatisch rebootet! Da:
>  /bin/aicmd pumaglued uimg   >>> switchandreboot <<<

 Siehst du die neue "and rebooting" Meldung aus dem Commit überhaupt? Also bist du im UIMG case des Scriptes?
Das alles ohne manuelles drücken von einem "Box neustarten" Button

**Comment by Prisrak1** [comment]
nur das hier ..
![Screenshot_20221206_184132](https://user-images.githubusercontent.com/45603872/205985074-722d40d8-fe52-4093-b218-f05b1284ad7a.png)

dann reboot und ..
![Screenshot_20221206_183304](https://user-images.githubusercontent.com/45603872/205983556-3000b408-43f6-41c7-9f57-d04c0a38952a.png)

wie schon geschrien, der Wechsel findet nicht statt. Es wird angezeigt: z.Z 1 dann -> 0. Das passiert jedoch nicht. Geprüft mit TCommander über FTP zugrif.

**Comment by fda77** [comment]
Aha, prima. Jedenfalls ist das Script im richtigen UIMG-Case

Da kommt das "changing ... and rebooting" her:  https://github.com/Freetz-NG/freetz-ng/blob/master/make/pkgs/mod/files/root/usr/mww/cgi-bin/exec.d/linux_fs_start.sh#L14

Wie du siehst wird in der nächsten Zeile dieses avm-"switchandreboot" ausgeführt 

...

Wie ist es denn wenn du diesen befehl im Terminal eingibst, rebootet die box dann selbst neu?

**Reply by fda77** [reply]
Vielleicht fehlt was vom environment, ich habs mal geladen: 396a16b24c6e0a3b795fdda5eabf18dd30e1edcb

**Comment by Prisrak1** [comment]
beim Wechsel der Partition sollte doch das in rot "deaktiviert" auf grün "aktiviert" änder. Tut es nich. Nur die Box wird gestartet.
![image](https://user-images.githubusercontent.com/45603872/206112437-b4b90873-44a7-49cb-8bce-49ec3d54eabf.png)
FTP sagt aktuell ist = 0
![image](https://user-images.githubusercontent.com/45603872/206113516-3315fbe9-c4ed-49db-8506-8ea8425f32b2.png)
über Putty geht´s zum Wechsel einwandfrei.

![Screenshot_20221207_081954](https://user-images.githubusercontent.com/45603872/206116002-11786495-4cf1-4a9d-b245-fca0a48af5f8.png)

**Comment by fda77** [comment]
Also das "aicmd" macht nicht was es soll wenn es durch das Webif ausgeführt wird? Also ein anderes Verhalten als im Teminal ausgerufen? Dann müsste jemand mit so einem Gerät herausfinden weshlab, vielleicht hilft strace

**Comment by Prisrak1** [comment]
welche Befehle werden benötigt?

**Comment by fda77** [comment]
Na wenn der "aicmd"-Aufruf im Terminal und im Webif verschieden Ergebnisse bringen passt irgend was nicht. Aufrufe innerhlab des Webifs haben öfter Probleme wegen den reduzirten Environment. Was jetzt genau das Problem ist müsste jemand mit so einer UIMG Hardware herausfinden/debuggen/ausprobieren. Ich kann da nichts machen

**Comment by Prisrak1** [comment]
Flashlog ..



Firmware extrahieren, Update vorbereitenFirmware-Archiv extrahieren ..../var/ ./var/tmp/ ./var/.packages ./var/install ./var/.config ./var/firmware-update.uimg ./var/version ./var/info.txt ./var/ewnw_check_install ./var/content ./var/signature ERLEDIGTAusführen des Firmware-Installationsskripts /var/install ...install: check and install new firmware ... OEM=avm ANNEX=Kabel testing acceptance for device Fritz_Box_HW267a ... OK - OEM avm is supported OK - accept this update for device Fritz_Box_HW267a ... testing acceptance for device Fritz_Box_HW267a done curr: 267.07.39  new: xx.07.39 debug: curr: 267.07.39 debug: new: "XX.07.39" major_currFWver=267 middle_currFWver=7 minor_currFWver=39 middle_newFWver=7 minor_newFWver=39 check Firmware Version: xx.07.39 DEBUG: 7 >= 7 DEBUG: 39 >= 39 Accept Firmware Version: xx.07.39 ERLEDIGT – Rückgabewert des Installationsskripts: 1 (INSTALL_SUCCESS_REBOOT)HINWEIS: Es gibt kein ausführbares Nach-Installationsskript.burnuimg: <<< 220 pumaglued ready. burnuimg: >>> uimg update burnuimg: <<< 227 /usr/sbin/avm_uimg_update starting. burnuimg: <<< 220 pumaglued ready. burnuimg: >>> uimg status burnuimg: <<< 225 update status follow. avm_uimg_update: OK APPCPU Kernel: type 2, size 9437184 APPCPU RootFS: type 3, size 45731840 NPCPU Kernel: type 8, size 2750464 NPCPU RootFS: type 9, size 16338944 GWFS: type 10, size 440320 UIMG: exit 0 STATUS exit 0 Das Gerät startet jetzt neu ...

**Comment by reimerp** [comment]
Ich hab erfolgreich per push_firmware flashen können, mich dann auch gewundert, das zwar das korrekte Image startet, aber die alte Version nicht angezeigt wurde. Zuvor hat ich schon vergeblich versucht, per bootloader zu switchen (das ging bei einer 6490 noch problemlos -- auch trotz der dokumentierten falschen Ausgabe im FTP)
Die Anzeige liegt scheinbar an _/usr/mww/cgi-bin/system_lfs.cgi_
Scheinbar gibt es auf der Box kein _/proc/avm_partitions_ mehr, das /proc/mtd liefert nur 5 Partitionen, zumindest ist das gesuchte "reserved" im Namen nicht dabei.
Mache ich mit die Mühe und analysiere die _/dev/mmcblk0p_ Devices, so
finde ich in p2 den aktiven Kernel und der Inhalt ist identisch zu p8 (gleiche AVM version in Reserve, und unmodifizierter Kernel)
In p3 liegt die aktuelle Freetz Version und in p9 die vorherige (geprüft durch mount, so wie es auch das obige script tut)
Fazit das Script kann die Partion per Name nicht finden, das Flash geht aber wie erwartet.
Firmware wechseln per GUI geht schief:
```
Toggling ...
changing 1 -> 0 ... and rebooting
failed.
```
aber es funktioniert per `aicmd`
Danach ist dann aber linux_fs_start=0 (geprüft in urloader und damit kommt die GUI gar nicht klar, angezeigt wird
"Momentan in linux_fs_start=1, deaktiviert beim nächsten Systemstart"
dann wird das aktuelle Image angezeigt
und dann
"Reserve in linux_fs_start=0, aktiviert beim nächsten Systemstart"
mit Unknown.
kann natürlich sein, das das mit obigem Problem zusammenhängt.

**Comment by fda77** [comment]
Hab noch nie eine mit Puma7 gehabt, daher basiert das ganz auf hörensagen. Die Qualität der Rückmeldungen ist recht unterschiedlich/unterirdisch, man weiss nie wie sehr man jemandem ernstnehmen kann. Aktuell hab ich 0 lust irgend jemandem 100 fragen zu stellen um herauszubekommen wie man das script am besten ändern würde. Kann sein dass AVM was mit einer anderen Kernelversion geändert hat.
Ich glaub "aicmd" wird genutzt, das sollte auch in diesem Verzeichnis liegen
meine lösung wäre: die ganze LFS anzeige für puma7 zu deaktivieren
